src/client/pages/Catalog.test.tsx

```diff
@@ -132,7 +120,9 @@ describe("<Catalog />", () => {
     );
 
     await waitFor(() => {
-      expect(screen.getAllByRole("listitem")).toHaveLength(2);
+      expect(
+        within(screen.getByTitle("Products")).getAllByRole("listitem"),
+      ).toHaveLength(2);
```

Comment:
Using http mocks led to more data being fetched, and required a more specific assertion.